src/photon/test/photon_tests.c

```diff
@@ -476,12 +515,77 @@ TEST task_multi_dependency_test(void) {
   PASS();
 }
 
+TEST start_kill_workers_test(void) {
+  /* Start some workers. */
+  int num_workers = 4;
```

Comment:
we should add a separate test (a stress test if you will) that tries to do this with 100 workers. I've tried it with 100 workers just now, and it wasn't working for me.